typing/typecore.ml

```diff
@@ -1402,7 +1535,12 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
         ) p2_variables;
         begin match p1, p2 with
         | None, None ->
-            if mode = Inside_or then raise Need_backtrack else split_or sp
+           let inside_nonsplit_or = match get_splitting_mode mode with
```

Comment:
Can you please put the `match ... with` on its own line?

Reply:
Are you sure you pushed then?

Reply:
Ah, I broke the `may_split, must_split` line last time, sorry. I'll break both.